### PR TITLE
Added deadline shift to -1 day to all availability results

### PR DESCRIPTION
**Pull request by mingazhev**

### CODE REVIEW

File: Api/Services/Accommodations/Availability/CounterpartySystemSettingsService.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Threading.Tasks;
+
+namespace HappyTravel.Edo.Api.Services.Accommodations.Availability
+{
+    public class CounterpartySystemSettingsService : ICounterpartySystemSettingsService
+    {
+        public async Task<CounterpartyAccommodationBookingSettings> GetAccommodationBookingSettings(int agentCounterpartyId)
+        {
+            // TODO implement different settings base on contract type
```
Comment (kirillta) [inline Api/Services/Accommodations/Availability/CounterpartySystemSettingsService.cs:10]: Will you do it in a separate task?

File: Api/Services/Accommodations/Availability/RoomContractSetPolicyProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                var roomContractDeadline = GetShiftedDeadline(roomContract.Deadline, checkInDate, shiftValue);
+                shiftedRoomContracts.Add(SetDeadline(roomContract, roomContractDeadline));
+            }
+
+            return new RoomContractSet(roomContractSet.Id, roomContractSet.Rate, roomContractSetDeadline, shiftedRoomContracts, roomContractSet.IsAdvancePurchaseRate);
+
+
+            static RoomContract SetDeadline(in RoomContract roomContract, Deadline roomContractDeadline) 
+                => new(
+                roomContract.BoardBasis,
```
Comment (kirillta) [inline Api/Services/Accommodations/Availability/RoomContractSetPolicyProcessor.cs:36]: Weird formatting here

File: Api/Services/Accommodations/Availability/RoomContractSetPolicyProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            return roomContractSet
+                .Select(r => Process(r, checkInDate, cancellationPolicyProcessSettings))
+                .ToList();
+        }
+        
+
+        public static RoomContractSet Process(RoomContractSet roomContractSet, DateTime checkInDate,
+            CancellationPolicyProcessSettings cancellationPolicyProcessSettings)
+        {
+            var shiftValue = cancellationPolicyProcessSettings.PolicyStartDateShift;
```
Comment (kirillta) [inline Api/Services/Accommodations/Availability/RoomContractSetPolicyProcessor.cs:22]: I just thought: shift must not be negative

File: Api/Services/Accommodations/Availability/RoomContractSetPolicyProcessor.cs (hunk 2 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Availability/RoomContractSetPolicyProcessor.cs:36]: Fixed

File: Api/Services/Accommodations/Availability/CounterpartySystemSettingsService.cs (hunk 1 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Availability/CounterpartySystemSettingsService.cs]: Added a task number

File: Api/Services/Accommodations/Availability/RoomContractSetPolicyProcessor.cs (hunk 3 above)
Comment (mingazhev) [reply Api/Services/Accommodations/Availability/RoomContractSetPolicyProcessor.cs:22]: Added such check

**Review by kirillta — APPROVED** [review]

**Pull request merged**